### ISSUE TITLE
Error preparing file: cypress/support/e2e.js

### ISSUE THREAD

**Issue by Strezzo06** [uktrade/cypress-image-diff#198]
I updated to v.2.1.2 and i found error about file: cypress/support/e2e.js
<img width="1117" alt="Screen Shot 2023-12-19 at 10 48 21 AM" src="https://github.com/uktrade/cypress-image-diff/assets/60592070/6d30a3f3-6bc4-4786-a69e-30274f13abe3">

my config file path: cypress/support/e2e.js
<img width="598" alt="Screen Shot 2023-12-19 at 10 54 25 AM" src="https://github.com/uktrade/cypress-image-diff/assets/60592070/d5de5609-74da-4075-bf2b-188fe0651455">

Cypress Version 12.9.0
Javascript
Mac OS 12.0.1

**Comment by kien-ht** [comment]
Hi, can you provide more detail context around your settings such as package.json, node version and project structure please?

**Comment by Strezzo06** [comment]
> Hi, can you provide more detail context around your settings such as package.json, node version and project structure please?

I run on docker
Docker File
<img width="675" alt="Screen Shot 2023-12-19 at 11 48 43 AM" src="https://github.com/uktrade/cypress-image-diff/assets/60592070/797ed49d-2d13-4e7e-9c30-6ba106e3f8fc">

I've truncated some parts in the package.json file that I use for better readability, as the file is very long.
`"scripts": {
"cypress": "cypress open",
"cy:stg:lcv:headless": "bash -c 'source .env.local-lcv && yarn run prereport && cypress run --browser electron --env ENV=stg,WEB=lcv,CONTENTSTACK_ENV=public-stg,CONTENTSTACK_API_KEY=${CONTENTSTACK_API_KEY},CONTENTSTACK_ACCESS_TOKEN=${CONTENTSTACK_ACCESS_TOKEN} --config baseUrl=https://pickup-stg.isuzu-tis.com/' && yarn run postreport",
"prereport:junit": "yarn run delete:reports:junit",
"postreport:junit": "yarn run combine:reports:junit",
"prereport:mochawesome": "yarn run delete:reports:mochawesome",
"postreport:mochawesome": "yarn run combine:reports:mochawesome && npx marge mochawesome.json",
"prereport": "yarn run prereport:mochawesome && yarn run prereport:junit",
"postreport": "yarn run postreport:mochawesome && yarn run postreport:junit",
},
	"devDependencies": {
		"@tripetch-isuzu/gitleaks": "^1.0.2",
		"cy-mobile-commands": "^0.3.0",
		"cypress": "12.9.0",
		"cypress-image-diff-html-report": "^2.0.2",
		"cypress-image-diff-js": "^2.1.2",
		"cypress-mochawesome-reporter": "^3.4.0",
		"cypress-multi-reporters": "^1.6.3",
		"cypress-ntlm-auth": "^4.1.2",
		"cypress-real-events": "^1.10.3",
		"cypress-recurse": "^1.29.0",
		"cypress-visual-regression": "^3.0.0",
		"cypress-wait-until": "^1.7.2",
		"del": "^7.0.0",
		"eslint": "7",
		"eslint-config-prettier": "^8.3.0",
		"eslint-plugin-cypress": "^2.12.1",
		"husky": "^8.0.3",
		"junit-report-merger": "^6.0.1",
		"lodash": "^4.17.21",
		"mocha": "^10.2.0",
		"mocha-junit-reporter": "^2.2.0",
		"mochawesome": "^7.1.3",
		"mochawesome-merge": "^4.3.0",
		"mochawesome-report-generator": "^6.2.0",
		"playwright-webkit": "^1.35.1",
		"prettier": "2.8.8"
	}`

Project structure
<img width="262" alt="Screen Shot 2023-12-19 at 11 45 09 AM" src="https://github.com/uktrade/cypress-image-diff/assets/60592070/882c2e77-637c-4958-9d3d-e90d089e06ed">
<img width="262" alt="Screen Shot 2023-12-19 at 11 45 20 AM" src="https://github.com/uktrade/cypress-image-diff/assets/60592070/b99e67dc-fdb6-4bf0-869e-de767f204d62">

**Comment by kien-ht** [comment]
This looks like "exports" field is not properly parsed by Webpack. What Webpack version are you using now?

**Comment by Lisa-Danielle** [comment]
I'm having this same issue!

**Comment by Strezzo06** [comment]
> This looks like "exports" field is not properly parsed by Webpack. What Webpack version are you using now?

I run pass after upgrade cypress from 12.9.0 to version 13.0.0
I think cypress-image-diff-js v2.1.2 not compatible with cypress version 12.9.0

**Comment by PippoRaimondi** [comment]
Let's leave this one open for a while until we figure out, good that we have a workaround but equally we want to ensure this can run on supported older versions...

**Comment by kien-ht** [comment]
> > This looks like "exports" field is not properly parsed by Webpack. What Webpack version are you using now?
> 
> I run pass after upgrade cypress from 12.9.0 to version 13.0.0 I think cypress-image-diff-js v2.1.2 not compatible with cypress version 12.9.0

Thank you for your feedback. This actually helps me find out the root of the problem. [This is indeed the problem with `exports` field in `package.json` not regconised by Webpack 4 and older](https://github.com/webpack/webpack/issues/9509).

Cypress uses Webpack under the hood for bundling JavaScript. As I was looking through the Cypress release notes, I found out that [Cypress v12.17.4](https://docs.cypress.io/guides/references/changelog#12-17-4) has upgraded Webpack from v4 to 5, and that if you use Cypress v12.17.4 and newer, you wouldn't encounter this problem.

To have better backward compatibility support, I created a pull request for this. We can close this issue for now, although we can leave it open for a while to ensure we have it right this time.
